File: data/compilers_comparation/src/examples/second_example/src/func51.c
#include "second_example.h" 
array_t* func51(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array59;
   if (pCounter > 0) {
      array59 = vars->data[--pCounter];
      array59->refC++;
      DEBUG_COPY(array59->id);
   } else {
      array59 = (array_t*)malloc(sizeof(array_t));
      array59->size = 606;
      array59->refC = 1;
      array59->id = 59;
      array59->data = (unsigned int*)malloc(array59->size*sizeof(unsigned int));
      memset(array59->data, 0, array59->size*sizeof(unsigned int));
      DEBUG_NEW(array59->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array59;
   array_t* array60 = func54(&params0, rng(), loopsFactor);
   DEBUG_RETURN(array60->id);
   free(params0.data);
   unsigned int loop19 = 0;
   unsigned int loopLimit19 = (rand()%loopsFactor)/4 + 1;
   for(; loop19 < loopLimit19; loop19++) {
      for (int i = 0; i < array59->size; i++) {
         array59->data[i]++;
      }
   }
   for (int i = 0; i < array59->size; i++) {
      if (array59->data[i] == 88) { 
         return array59;
      }
   }
   array59->refC--;
   if(array59->refC == 0) {
      free(array59->data);
      free(array59);
      DEBUG_FREE(array59->id);
   }
   return array60;
}

File: data/multilang_comparison/program_nim/src/func51.nim
proc func51*(vars: ptr ArrayTParam, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   var array88: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array88 = vars.data[pCounter]
      inc array88.refC
      debugCopy(array88.id)
   else:
      array88 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array88.size = 804
      array88.refC = 1
      array88.id = 88
      array88.data = cast[ptr UncheckedArray[uint32]](alloc(array88.size * sizeof(uint32)))
      zeroMem(array88.data, array88.size * sizeof(uint32))
      debugNew(array88.id)
   var params0: ArrayTParam
   params0.size = 1
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   params0.data[0] = array88
   var array89: ptr ArrayT = func54(addr params0, getPath(), loopsFactor)
   debugReturn(array89.id)
   dealloc(params0.data)
   let loopLimit28 = (50) div (4) + 1
   for loop28 in 0 ..< loopLimit28:
      discard
      for i in 0 ..< array88.size:
         inc array88.data[i]
   for i in 0 ..< array88.size:
      if array88.data[i] == 99:
         return array88
   dec array88.refC
   if array88.refC == 0:
      dealloc(array88.data)
      dealloc(array88)
      debugFree(array88.id)
   return array89

